PowerEditor/src/winmain.cpp

```diff
@@ -21,6 +21,78 @@
 #include "verifySignedfile.h"
 #include "NppDarkMode.h"
 
+// Include that contain COM classes and IDLs
+// The COM interfaces that we are about here are only available if we specify Windows 10 or higher,
+// So we override the default _WIN32_WINNT that is set in the project file for this include.
+#if defined(_WIN32_WINNT) && (_WIN32_WINNT < _WIN32_WINNT_WIN10)
+#	undef _WIN32_WINNT
+#endif
+#define _WIN32_WINNT _WIN32_WINNT_WIN10
+#include <ShObjIdl.h>
+
+// MinGW and such have IDL headers that do not have IVirtualDesktopManager.
+// Thus, we need to provide the declarations in that case, and ideally
+// declarations that are syntactically valid for both MSVC _and_ MinGW-g++.
+// 
+// These are lifted from ShObjldl.h and ShObjldl_core.h
+#ifndef __IVirtualDesktopManager_INTERFACE_DEFINED__

```

Comment:
I'd suggest moving it to a separate header file and including here.

Reply:
Probably a good idea.

It's frustrating as it only exists because the MinGW headers are lacking these IDL definitions.